packages/plugin-inline/Inline.ts

```diff
@@ -0,0 +1,106 @@
+import { JWPlugin } from '../core/src/JWPlugin';
+import { RangeParams } from '../core/src/CorePlugin';
+import { Format } from './Format';
+import { InlineNode } from './InlineNode';
+
+export interface FormatParams extends RangeParams {
+    format: Format;
+}
+
+export class Inline extends JWPlugin {
+    commands = {
+        toggleFormat: {
+            handler: this.toggleFormat.bind(this),
+        },
+    };
+    commandHooks = {
+        setSelection: this.resetFormatCache.bind(this),
+    };
+    /**
+     * When apply format on a collapsed range, cache the calculation of the
+     * format the following property. This value is reset each time the range
+     * change in a document.
+     */
+    formatCache: Format[] = null;
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Apply the `format` to the selection.
+     *
+     * If the selection is collapsed, set the format on the selection so we know
+     * in the next insert which format should be used.
+     */
+    toggleFormat(params: FormatParams): void {
+        const format = params.format;
+        const range = this.editor.vDocument.selection.range;
+        if (range.isCollapsed()) {
+            if (!this.formatCache) {
+                this.formatCache = this.getCurrentFormats();
+            }
+            const index = this.formatCache.findIndex(f => f.name === format.name);
+            if (index !== -1) {
+                this.formatCache.splice(index, 1);
+            } else {
+                this.formatCache.push(format.shallowDuplicate());
+            }
+        } else {
+            const selectedInlines = range.selectedNodes(InlineNode);
+
+            // If every char in the range has the format `format.name`, remove
+            // the format for all of them.
+            const allHaveFormat = selectedInlines.every(inline => {
+                return !!inline.formats.find(f => f.name === format.name);
+            });
+            if (allHaveFormat) {
+                selectedInlines.forEach(inline => {
+                    const index = inline.formats.findIndex(f => f.name === format.name);
+                    const matchingFormat = inline.formats[index];
+                    if (matchingFormat.attributes && matchingFormat.attributes.size) {
+                        matchingFormat.attributes.forEach((value: string, key: string) => {
+                            if (!inline.attributes) {
+                                inline.attributes = new Map<string, string>();
```

Comment:
That's the magic which is going to keep the `b` classes on the `span` when rendered, right ?